cli/wallet/nep5.go

```diff
@@ -28,6 +49,77 @@ func newNEP5Commands() []cli.Command {
 	}
 }
 
+func getNEP5Balance(ctx *cli.Context) error {
+	wall, err := openWallet(ctx.String("path"))
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	defer wall.Close()
+
+	addr := ctx.String("addr")
+	addrHash, err := address.StringToUint160(addr)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("invalid address: %v", err), 1)
+	}
+	acc := wall.GetAccount(addrHash)
+	if acc == nil {
+		return cli.NewExitError(fmt.Errorf("can't find account for the address: %s", addr), 1)
+	}
+
+	var token *wallet.Token
+	name := ctx.String("token")
+	if name != "" {
+		token, err = getMatchingToken(wall, name)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+	}
+
+	gctx, cancel := getGoContext(ctx)
+	defer cancel()
+
+	c, err := client.New(gctx, ctx.String("rpc"), client.Options{})
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	balances, err := c.GetNEP5Balances(addrHash)
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	for i := range balances.Balances {
+		asset := balances.Balances[i].Asset
+		if name != "" && !token.Hash.Equals(asset) {
+			continue
+		}
+		fmt.Printf("TokenHash: %s\n", asset)
+		fmt.Printf("\tAmount : %s\n", balances.Balances[i].Amount)
+		fmt.Printf("\tUpdated: %d\n", balances.Balances[i].LastUpdated)
+	}
+	return nil
+}
+
+func getMatchingToken(w *wallet.Wallet, name string) (*wallet.Token, error) {
+	var token *wallet.Token
+	var count int
+	for _, t := range w.Extra.Tokens {
+		if t.Name == name || t.Symbol == name || t.Address == name || t.Hash.StringLE() == name {
+			if count == 1 {
+				printTokenInfo(token)
+				printTokenInfo(t)
+				return nil, errors.New("multiple matching tokens found")
+			}
+			count++
+			token = t
+		}
+	}
+	if count == 0 {
+		return nil, errors.New("import token before usage")
```

Comment:
I don't think this import requirement will make our users happy.

Reply:
What will make them happy though, is the ability to use abbreviations instead of the hashes.

Reply:
Or do you talk about allowing usage of hashes even for non-imported tokens?

Reply:
Yeah, I think there should be some ability to use tokens without importing. So import should only be done for some processing speedup, but not as a hard requirement.

Reply:
Fixed!